app/src/Menu.cpp

```diff
@@ -15,9 +18,8 @@ void Menu::draw() const
 
 void Menu::chooseAction()
 {
-    int value;
-    std::cin >> value;
-    _action = MenuAction(value);
+    InputController inputChoice;
```

Comment:
Use the InputController as a std::shared_ptr field of Menu class. Call it as _inputController